python/revng/scripting/project.py

```diff
@@ -0,0 +1,67 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from typing import IO, Tuple
+
+import yaml
+
+from revng.model import Binary
+
+
+class Project(ABC):
+    @abstractmethod
+    def fetch(self, *args, **kwargs) -> bytes:
+        pass
+
+    @abstractmethod
+    def fetch_many(self, *args, **kwargs) -> dict:
+        pass
```

Comment:
What's the difference between this and `fetch`?

Reply:
We had this example in the [pad](https://pad.rev.ng/jDodmNlYQMSZ_raFTrrGvw#Expected-usage). The difference is how you get the artifact. But I can unify both methods into one.